providers/gcp/resources/discovery.go

```diff
@@ -153,6 +184,50 @@ func discoverOrganization(conn *connection.GcpConnection, gcpOrg *mqlGcpOrganiza
 	return assetList, nil
 }
 
+func discoverFolder(conn *connection.GcpConnection, gcpFolder *mqlGcpFolder) ([]*inventory.Asset, error) {
+	assetList := []*inventory.Asset{}
+
+	if stringx.ContainsAnyOf(conn.Conf.Discover.Targets, DiscoveryAll, DiscoveryAuto, DiscoveryProjects) {
+		projects := gcpFolder.GetProjects()
+		if projects.Error != nil {
+			return nil, projects.Error
+		}
+
+		projectList := projects.Data.GetList() // resolve all projects including nested
```

Comment:
don't remember this fully but wasn't it possible to have a folder inside a folder in GCP? If that's the case, then I guess with this code we don't discover nested folders, but only nested projects

Reply:
That is ok, we can tackle that later.